external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSource.scala

```diff
@@ -124,9 +124,14 @@ private[kafka010] class KafkaSource(
 
   override def getDefaultReadLimit: ReadLimit = {
     if (minOffsetPerTrigger.isDefined && maxOffsetsPerTrigger.isDefined) {
-      ReadLimit.compositeLimit(Array(
-        ReadLimit.minRows(minOffsetPerTrigger.get, maxTriggerDelayMs),
-        ReadLimit.maxRows(maxOffsetsPerTrigger.get)))
+      val minOffsets = minOffsetPerTrigger.get
+      val maxOffsets = maxOffsetsPerTrigger.get
+      if (minOffsets > maxOffsets) {
+        throw new AnalysisException(s"The value of minOffsetPerTrigger($minOffsets) is higher " +
+          s"than the maxOffsetsPerTrigger($maxOffsets)")
+      }
```

Comment:
Should we check it in `validateBatchOptions` and `validateBatchOptions`?

Reply:
Either is fine, but having this in validateBatchOptions looks to be more coherent.

Reply:
make sense, I moved the check to `validateGeneralOptions`

Reply:
Done in f7eae4d.